openedx/core/djangoapps/user_api/management/tests/test_email_opt_in_list.py

```diff
@@ -393,7 +412,11 @@ def _assert_output(self, output, *args, **kwargs):
         for user, course_id, opt_in_pref in args:
             self.assertIn({
                 "email": user.email.encode('utf-8'),
-                "full_name": user.profile.name.encode('utf-8'),
+                "full_name": (
+                    user.profile.name.encode('utf-8')
+                    if hasattr(user, 'profile')
```

Comment:
Yeah, I see the conundrum. In general, I'm not crazy about conditionals in tests, since they can mask errors. Something else you can try is adding the user's full name to the args tuple